dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -70,3 +70,17 @@ properties:
     description: |
       This is a number >= 1 that is used to determine how many revolutions
       were done based on the current counter's value.
+
+  st,encoder-mode:
+    type: int
+    description: |
+      Encoder counting mode
+      MODE_X2_TI1 Counter counts up/down on TI2FP2 edge depending on TI1FP1 level
+      MODE_X2_TI2  Counter counts up/down on TI1FP1 edge depending on TI2FP2 level
+      MODE_X4_TI12 Counter counts up/down on both TI1FP1 and TI2FP2 edges
+      depending on the level of the other input
+    required: true
```

Comment:
Since this is a breaking change for existing users, please update `doc/releases/migration-guide-3.5.rst` (as a required change)

Reply:
> * Quadrature decoder STM32 driver now has a new property :dtcompatible:`st,encoder-mode` which is required.
  The old behavior is the same as setting the property as :c:enumerator:`MODE_X2_TI1`.
  
Will this kind of text suffice?

Reply:
Perfect!

Reply:
This change should be squashed into the commit that introduces the `st,encoder-mode` property